torchtune/training/quantization.py

```diff
@@ -142,15 +153,20 @@ def get_quantizer_mode(quantizer: Optional[Callable]) -> Optional[str]:
     Currently supported:
 
     - :class:`~torchao.quantization.quant_api.Int8DynActInt4WeightQuantizer`: "8da4w" (requires ``torch>=2.3.0``)
-    - :class:`~torchao.quantization.prototype.qat.Int8DynActInt4WeightQATQuantizer`: "8da4w-qat" (requires ``torch>=2.4.0``)
+    - :class:`~torchao.quantization.qat.Int8DynActInt4WeightQATQuantizer`: "8da4w-qat" (requires ``torch>=2.4.0``)
 
     Args:
         quantizer (Optional[Callable]): A callable object that implements the `quantize` method.
 
     Returns:
         Optional[str]: The quantization mode.
     """
-    return _quantizer_to_mode.get(type(quantizer), None)
+    mode = _quantizer_to_mode.get(type(quantizer), None)
+    if mode is not None and "module-swap" in mode:
+        warn(
+            "*QuantizerModuleSwap is deprecated. Please use the version without 'ModuleSwap' instead"
```

Comment:
nit: should be able to just give the exact classes in the warn message using e.g. `f"{quantizer.__class__.__name__}"` (or something like that)

Reply:
hmm I don't think that's possible given we just assign the classes today, e.g.:
```
>>> class A: pass
... 
>>> B = A
>>> A().__class__.__name__
'A'
>>> B().__class__.__name__
'A'
```